### ISSUE TITLE
Support fmf keys => result IDs transition in a declarative manner, without hardcoding of keys

### ISSUE THREAD

**Issue by happz** [teemtee/tmt#1493]
With https://github.com/teemtee/tmt/pull/1473, tmt will expose some of the `extra-*` keys as result IDs.

```python
            self.ids = {
                tmt.identifier.ID_KEY: test.node.get(tmt.identifier.ID_KEY),
                'extra-nitrate': test.node.get('extra-nitrate'),
                'extra-task': test.node.get('extra-task')}
```

The issue is, why just these two keys and not `extra-foo`? It would be better if `Result` would be aware of `nitrate` or "task" at all, and what keys to expose as result IDs would be given in a more open manner, e.g. for plugins to register their own, or as a list to update, property to call, etc. In general, without hardcoding the list of keys into a base class.

**Comment by happz** [comment]
I disagree, I have something else on my mind, even less intrusive than a lust of keys in tmt core.

**Comment by psss** [comment]
Seems the issue was closed as part of merging of the pull request.

**Comment by psss** [comment]
@happz, is this one still relevant?

**Comment by happz** [comment]
> [@happz](https://github.com/happz), is this one still relevant?

Unfortunately it is still relevant:

```
# Extra keys used for identification in Result class
EXTRA_RESULT_IDENTIFICATION_KEYS = ['extra-nitrate', 'extra-task']
```

**Comment by happz** [comment]
Still something that would be nice to have.
